### ISSUE TITLE
Handling Spaces and Line Breakers in HTML input

### ISSUE THREAD

**Issue by amirhbakan** [oleast/contentful-rich-text-html-parser#188]
When I pass the following HTML:
```
<h2>Explore the Wonders of the Ocean</h2>

<p>The ocean is a vast and mysterious place, teeming with life and beauty. From the smallest plankton to the largest whales, the marine ecosystem is incredibly diverse.</p>

```

It converts it to an invalid Contentful rich text and creates blocks with node type text as root because of \n characters. it return error when you pass it to Contentful management API.

<img width="559" alt="Contentful line breaker error" src="https://github.com/oleast/contentful-rich-text-html-parser/assets/12600125/039e4a93-6048-4ed0-9156-7e01de47d907">

I solved it by removing all the spaces and line breakers between HTML tags, at the start and end of my input and I believe library should have an option to choose between ignoring spaces and line breakers(by minifying HTML) or converting them to valid `<br/>` tags behaviour.
Even if it's not needed to have an extra option, at least solve it by implementing one of options expressed above and then pass it to converter.

Steps to Reproduce:

- Pass the above HTML to the converter.
- Observe the resulting Contentful rich text structure.
Expected Behaviour: The HTML should be converted to a valid Contentful rich text structure without creating blocks with node type text as root. when I pass this to Contentful API, it returns the error which is invalid block type because of text nodeType.

Actual Behavior: The HTML is converted to an invalid Contentful rich text structure with blocks having node type text as root due to \n characters. when I pass it Contentful management API, it return error said the block with node type of `text` is invalid.

Suggested Solution: Add an option to ignore or convert spaces and line breakers to something works or minify the HTML before converting it to the Contentful rich text data structure.

**Comment by oleast** [comment]
Good catch, and thanks for the detailed issue, @amirhbakan!

If I'm catching the issue correctly here the problem is just the `text`-nodes at the root of the document. Which means any text or whitespace at root of the original HTML-input would result in the same problem? E.g.:
``` html
<h2>Heading</h2>
Text outside an element
<p>Text inside an element</p>
```

My understading is that we can't turn all `\n` characters into `<br />`, because newlines are a valid part of text inside an HTML-document, and does not necesseraly serve the same purpose as `<br />`. If that behavior is intended by a user of the library, it should be possible to do it with a custom converter function.

Since the problem we're dealing with is text and whitespace that is at the root of an HTML-document (outside any HTML-tag), we're actually dealing with invalid HTML? But since we're dealing with what might be partial HTML-snippets, we'll have to handle it anyway. 

It would be possible to just add an option to `ingoreWhitespaceAtDocumentRoot` or something like that. But it seems like a very specific solution. Could also just do that by default without an option, since the resulting document will be invalid anyway. That would however not solve the problem with other text nodes at the document root.

I'll have to think about a solution a bit, thanks for the input!

**Comment by oleast** [comment]
Ok, since we're specifically dealing with Contentful here, maybe we could just wrap all text nodes at the root level of the document in a paragraph block?

If we reverse the situation, the Contentful Rich Text editor would never let you write `text`-nodes at the root level, they'll always be wrapped in a `paragraph`-node.

**Comment by oleast** [comment]
Hello, @amirhbakan!

It has taken some time to get back to this. I've implemented a few changes that I hope will solve this problem, they are currently available in #242. Do you think those changes will solve your problems?
